### ISSUE TITLE
UIA: crash at very end of buffer

### ISSUE THREAD

**Issue by codeofdusk** [microsoft/terminal#7771]
In the NVDA Python console (NVDA+control+z):

``` python console
>>> from NVDAObjects.UIA import UIATextInfo
>>> ti=UIATextInfo(nav, "last")
>>> ti.move("character", 1)  # conhost crashes
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "NVDAObjects\UIA\__init__.pyc", line 752, in move
  File "comtypesMonkeyPatches.pyc", line 26, in __call__
_ctypes.COMError: (-2147220991, 'An event was unable to invoke any of the subscribers', (None, None, None, 0, None))
```

Note: fixing this bug may result in strange behaviour from NVDA (see [here](https://github.com/nvaccess/nvda/blob/master/source/NVDAObjects/UIA/winConsoleUIA.py#L53), around line 53). I'll remove the offending workaround once a fix is merged to master.

I suspect #6986 would fix this bug, but I think the root cause of this crash should be investigated before then. Cc @carlos-zamora.

**Comment by DHowett** [comment]
This is with `main`, not Carlos's PR #7770, yes?

**Comment by codeofdusk** [comment]
Correct.

**Comment by carlos-zamora** [comment]
@codeofdusk what does `UIATextInfo(nav, "last")` do?

**Comment by codeofdusk** [comment]
See [here](https://github.com/nvaccess/nvda/blob/master/source/NVDAObjects/UIA/__init__.py#L292) (starting at around line 292), but in summary it gets a document range degenerate at end (i.e. moves start to where end is).
